### ISSUE TITLE
Is it possible not to suppress style errors inside source code?

### ISSUE THREAD

**Issue by swsms** [PyCQA/pylint#3792]
## Question

There is my case: we are developing an educational online platform for programmers and check their code using configured Pylint and some other linters. But some of advanced learners can just disable all checks for their code using special comments like below.

```python
def f(x):  # pylint: disable=R1710
    if x <= -2:
        return 1 - (x + 2) ** 2
    elif -2 < x <= 2:
        return -(x / 2)
    elif x > 2:
        return 1 + (x - 2) ** 2

```
Is it possible to fully ignore the `pylint: disable=R1710` (and any other code) comment using some kind of pylint settings? i.e. I want to see the warning R1710 in the linter's output although the code has the disabling comment. Unfortunately, I cannot find this in docs.

**Comment by AWhetter** [comment]
To any new contributors looking to implement this issue, the `pylint.lint.pylinter.PyLinter.process_tokens()` method is responsible for parsing pragmas (the `enable=` and `disable=` comments). We can add an check in that method that uses of a new option valid, which you can add to `PyLinter._make_options`, to skip calling https://github.com/PyCQA/pylint/blob/fb332490c2e59c232c7095dfe3462f8e081e2814/pylint/lint/pylinter.py#L749 and instead call `self.get_active_msgids(msgid)` so that we still validate that the error is valid.

**Comment by PCManticore** [comment]
Hey @swsms Thanks for the report. While this new option would make sense for your use case, it's something that we should not support in a general linter. I don't see a situation other than yours where this option would be useful. Maybe you need to subclass `PyLinter` class and disable this behaviour? This way you control the tool without relying on upstream modifications from us to support the use case.
